smart_contracts/contract/src/contract_api/storage.rs

```diff
@@ -335,3 +335,72 @@ pub fn disable_contract_version(
 
     api_error::result_from(result)
 }
+
+/// Creates new [`URef`] that points to a context-local partition of global state.
+/// Optionally you can pass a [`str`] to automatically put this dictionary under named keys.
+pub fn new_dictionary(key_name: &str) -> Result<URef, ApiError> {
+    if key_name.is_empty() || runtime::has_key(key_name) {
+        return Err(ApiError::InvalidArgument);
+    }
+
+    let value_size = {
+        let mut value_size = MaybeUninit::uninit();
+        let ret = unsafe { ext_ffi::casper_new_dictionary(value_size.as_mut_ptr()) };
+        api_error::result_from(ret)?;
+        unsafe { value_size.assume_init() }
+    };
+    let value_bytes = runtime::read_host_buffer(value_size).unwrap_or_revert();
+    let uref: URef = bytesrepr::deserialize(value_bytes).unwrap_or_revert();
+    runtime::put_key(key_name, Key::from(uref));
+    Ok(uref)
+}
+
+/// Reads data from dictionary pointed at by [`URef`].
+pub fn dictionary_get<K: ToBytes, V: CLTyped + FromBytes>(
+    uref: URef,
+    key: K,
+) -> Result<Option<V>, bytesrepr::Error> {
```

Comment:
I think the comment is unhelpful. Maybe something like...
/// Retrieve value stored under `key` in the dictionary accessed by `dictionary_uref`.